src/project-generator.ts

```diff
@@ -51,266 +39,97 @@ export class ProjectGenerator {
      * @param {string} authToken as PAT or Oauth Token
      */
     constructor(private owner: string, private repo: string, private authToken: string) {
-        this.requestConfig = {
-            headers: {
-                'Content-Type': 'application/json',
-                Accept: 'application/vnd.github+json',
-                Authorization: `Bearer ${this.authToken}`,
-            },
-        };
-        this.repositoryPath = `${GITHUB_API_URL}/${this.owner}/${this.repo}`;
+        this.gitRequestHandler = new GitRequestHandler(this.owner, this.repo, this.authToken);
     }
 
     /**
      * @param {string} codeSnippet Base64 encoded playground code snippet.
      * @param {string} appName Name of the VehicleApp.
+     * @param {string} vspecPayload Base64 encoded Vspec payload.
      * @throws {ProjectGeneratorError}
      */
-    public async run(codeSnippet: string, appName: string): Promise<number> {
+    public async runWithPayload(codeSnippet: string, appName: string, vspecPayload: string): Promise<number> {
         try {
-            await this.generateRepo();
+            let decodedVspecPayload = JSON.parse(decode(vspecPayload));
+            await this.gitRequestHandler.generateRepo();
             // Delay is introduced to make sure that the git API creates
             // everything we need before doing other API requests
             await delay(MS_TO_WAIT_FOR_GITHUB);
-            await this.updateContent(appName, codeSnippet);
+            await this.updateContent(appName, codeSnippet, decodedVspecPayload);
             return StatusCodes.OK;
         } catch (error) {
             throw error;
         }
     }
 
-    private async generateRepo(): Promise<number> {
+    /**
+     * @param {string} codeSnippet Base64 encoded playground code snippet.
+     * @param {string} appName Name of the VehicleApp.
+     * @param {string} vspecUri Root URI of Vspec Files.
```

Comment:
You are calling `decode` on `vspecUri`. You need to mention that this parameter is also expected to be base64 encoded.